### ISSUE TITLE
Code blocks have too much spacing between lines

### ISSUE THREAD

**Issue by jwflory** [unicef/inventory-hugo-theme#62]
# Summary

Hugo's [`highlight` syntax highlighter](https://gohugo.io/functions/highlight/) has too much whitespace between lines, and lines of code do not match with line numbers.

# How to reproduce?

1. Create a block of code inside a content file using `{{< highlight >}}`
2. Run `hugo serve`
3. Go to the page with the block of code

# Expected behavior

A code block appears with no spacing between lines. Each line of code matches up to the line numbering in the highlighter sidebar.

# Actual behavior

There are two new lines between each individual line of code. The lines of code do not match up to the lune numbering in the highlighter sidebar.

**Screenshots**:

![Screenshot of a code block with too much spacing between the lines. Lines of code do not match up with numbered lines.](https://user-images.githubusercontent.com/4721034/157731410-5a24e03c-8d09-4acc-b1b4-ddc9e3bfe164.png "Screenshot of a code block with too much spacing between the lines. Lines of code do not match up with numbered lines.")

# Other details

An example of how we currently use the [`highlight` function](https://gohugo.io/functions/highlight/) in our site (see it [_here_](https://github.com/unicef/inventory/blob/main/content/meta/inventory/maintainers-guide.en.adoc#example-git-in-command-line)):

```sh
{{< highlight bash "linenos=table,hl_lines=8 15-17,linenostart=1" >}}
git submodule update --remote --rebase
cd themes/inventory/
git fetch
git checkout <latest commit hash>
cd ../../
git add themes/inventory
git commit --signoff
{{< / highlight >}}
```

**Comment by BeeBombshell** [comment]
Hi @jwflory, can I work on this issue? 😊

**Comment by jwflory** [comment]
@BeeBombshell Sure, it's yours 👍🏻 The links in the original comment will help you get started. Ask here or in the Matrix room if you have questions or face difficulties while working on this.

**Comment by BeeBombshell** [comment]
Hi @jwflory, I made some changes in the `config.yaml` file, and hence the `highlight` function works fine. However, if a user manually changes the `lineNumbersInTable` parameter, then the lines of code do not match up to the line numbering.

After I changed the default values in the `config.yaml` file, the code looks like this:

<img width="446" alt="image" src="https://user-images.githubusercontent.com/53828745/158357696-c82b4b89-03d8-46bf-9ef9-5b49ad6595a7.png">

**Comment by BeeBombshell** [comment]
Further, I was looking at how to reduce the spacing between the code lines. There is no parameter in the `highlight` function to deal with this as such. I found a solution in the Hugo Forums suggesting some changes in the newline characters in the MD file.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/53828745/158359773-de0504e3-4f7b-4e2e-ad47-7b316df01ac4.png">

Link for the Hugo Forum: https://discourse.gohugo.io/t/built-in-highlight-function-adds-whitespace-new-lines/19058/13

Kindly take a look and tell me if I should go ahead and try the same... 😊

**Comment by BeeBombshell** [comment]
@jwflory I tried changing the parameters in different combinations. Also, I tried using the `highlight` block without the inline CSS style which led to the following outcome...

<img width="456" alt="image" src="https://user-images.githubusercontent.com/53828745/158410102-b8894621-7649-4372-ba45-caee1ffb9ef7.png">

However, the problem is that without the inline CSS, the code block doesn't recognize the keywords of the input language. Even if we use an external CSS file, we'd still have to define the previous parameters which would lead to the same outcome without proper inline spacing.

I tried replacing the CRLF line endings with LF on my local machine but that doesn't help. So, I've made some minor changes in the `config.yaml` file which solves the alignment problem for now... Should I go ahead and submit a PR for the same? 😊
